x509-ocsp/src/ext.rs

```diff
@@ -0,0 +1,221 @@
+//! OCSP Extensions
+
+use alloc::vec::Vec;
+use const_oid::{
+    db::rfc6960::{
+        ID_PKIX_OCSP_ARCHIVE_CUTOFF, ID_PKIX_OCSP_CRL, ID_PKIX_OCSP_NONCE,
+        ID_PKIX_OCSP_PREF_SIG_ALGS, ID_PKIX_OCSP_RESPONSE, ID_PKIX_OCSP_SERVICE_LOCATOR,
+    },
+    AssociatedOid,
+};
+use der::{
+    asn1::{GeneralizedTime, Ia5String, ObjectIdentifier, OctetString, Uint},
+    Sequence, ValueOrd,
+};
+use spki::AlgorithmIdentifierOwned;
+use x509_cert::{
+    ext::{pkix::AuthorityInfoAccessSyntax, Extension},
+    impl_newtype,
+    name::Name,
+};
+
+#[cfg(feature = "rand_core")]
+use rand_core::CryptoRngCore;
+
+/// Trait to be implemented by extensions to allow them to be formated for x509 OCSP
+/// requests/responses.
+pub trait AsExtension: AssociatedOid + der::Encode {
```

Comment:
Is there a reason we can't pull [`x509_cert::ext::AsExtension`](https://docs.rs/x509-cert/latest/x509_cert/ext/trait.AsExtension.html) ?

Besides the `const` that looks very similar.

Reply:
It can be used. The extra subject/extensions arguments for judging criticality didn't seem relevant at all to OCSP though.

Reply:
If we change x509_cert::ext::AsExtension to accepts subject as Option<&Name>, it'd be easier. Otherwise, I think I'd have to build an empty Name just to ignore it later.

Reply:
or a `Name::default()` :D